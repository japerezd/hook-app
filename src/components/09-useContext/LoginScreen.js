import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    
    // 1. obtener referencia al userContext
    const { setUser} = useContext(UserContext)
    // 2. setUser
    // 3. {id:1234, name: 'jorge'}
    const user = {
        id: 1234,
        name: 'Jorge'
    }
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className="btn btn-primary"
                onClick={() => setUser(user)}
            >Login</button>
        </div>
    )
}
