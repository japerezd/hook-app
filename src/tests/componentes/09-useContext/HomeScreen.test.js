import { mount, shallow } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen />', () => {

    const user = {
        name:'Jorge',
        email: 'jorge@gmail.com'
    }

    // Como mi HomeScreen se le pasa un user, se debe de pasarlo mediante
    // <UserContext.Provider value = {} />

    // Se debe usar mount en lugar de shallow para que pueda renderizar el homescreen
    //y nosolo el high order component
    const wrapper = shallow(
        <UserContext.Provider value = {{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    
    );

    test('debe mostrar el componente ', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
