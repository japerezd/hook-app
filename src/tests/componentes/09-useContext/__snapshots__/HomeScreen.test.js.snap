// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe mostrar el componente  1`] = `<HomeScreen />`;
